java-checks/src/main/java/org/sonar/java/checks/HardcodedIpCheck.java

```diff
@@ -48,21 +57,28 @@ public void visitLiteral(LiteralTree tree) {
     if (tree.is(Tree.Kind.STRING_LITERAL)) {
       IP.reset(tree.value());
       if (IP.matches()) {
+        ArrayList<Integer> octets = new ArrayList<Integer>(4);
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Replace the type specification in this constructor call with the diamond operator ("<>"). [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS2293)